src/Deserialization.cpp

```diff
@@ -0,0 +1,1358 @@
+#include "Deserialization.h"
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+#include "FindCalls.h"
+#include "Func.h"
+#include "Function.h"
+#include "IR.h"
+#include "Schedule.h"
+#include "halide_ir_generated.h"
+
+namespace Halide {
+namespace Internal {
+class Deserializer {
+public:
+    Deserializer() = default;
+
+    Deserializer(const std::unordered_map<std::string, Parameter> &params)
```

Comment:
A single-argument constructor should usually be tagged as `explicit`. (If it is deliberately not explicit because you want implicit conversion, then add a `/*not-explicit*/` as a comment.)